strawberryfields/tdm/tdmprogram.py

```diff
@@ -0,0 +1,232 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module implements the :class:`.TDMProgram` class which acts as a representation for time-domain quantum circuits.
+"""
+# pylint: disable=too-many-instance-attributes,attribute-defined-outside-init
+
+from operator import itemgetter
+from math import ceil
+import strawberryfields as sf
+from strawberryfields import ops
+from strawberryfields.parameters import par_is_symbolic
+
+
+def shift_by(l, n):
+    """Convenience function to shift a list by a number of steps.
+
+    If ``n`` is positive it shifts to the left.
+
+    Args:
+        l (list): list to be shifted
+        n (int): size of the shift
+    """
+    return l[n:] + l[:n]
+
+
+def get_modes(cmd, q):
+    """Returns the registers from q required by a command"""
+    reg_getter = itemgetter(*[r.ind for r in cmd.reg])
+    modes = reg_getter(q)
+
+    if not isinstance(modes, tuple):
+        modes = (modes,)
+
+    return modes
+
+
+def validate_measurements(circuit, N):
+    """Validate the TDM program measurements are correct"""
+    spatial_modes = 0
+    measurement_reg = []
+
+    for cmd in circuit:
+        if isinstance(cmd.op, ops.Measurement):
+            modes = [r.ind for r in cmd.reg]
+            measurement_reg.extend(modes)
+            spatial_modes += len(modes)
+
+    if not spatial_modes:
+        raise ValueError("Must be at least one measurement.")
+
+    if spatial_modes is not len(N):
+        raise ValueError("Number of measurements operators must match number of spatial modes.")
```

Comment:
Sorry if I make this comment too often but the typo is still there. I believe it should say "measurement operators" instead of "measurement**S** operators".

Reply:
Corrected now :)